### PR TITLE
new_audit: unsized-images reland

### PR DESCRIPTION
**Pull request by connorjclark**
Undoing revert from #11317 and relanding...

Adding [just one change](https://github.com/GoogleChrome/lighthouse/pull/11340/commits/6449d8b80c2ac9470c5787dc9f37b2761b45393d): skipping fetching the source rules in `image-elements` if the image is from css (background-image). That sped up some of the slowest sites by ~50s. See https://github.com/GoogleChrome/lighthouse/pull/11336

ref https://github.com/GoogleChrome/lighthouse/issues/11289

### CODE REVIEW

File: lighthouse-core/audits/unsized-images.js [hunk 1]
```diff
@@ -67,31 +67,34 @@ class SizedImages extends Audit {
...
 
   /**
    * @param {LH.Artifacts} artifacts
    * @return {Promise<LH.Audit.Product>}
    */
   static async audit(artifacts) {
-    // CSS background-images are ignored for this audit.
-    const images = artifacts.ImageElements.filter(el => !el.isCss);
+    // CSS background-images & ShadowRoot images are ignored for this audit.
+    const images = artifacts.ImageElements.filter(el => !el.isCss && !el.isInShadowDOM);
```
Comment (connorjclark) [inline lighthouse-core/audits/unsized-images.js:90]: only change

File: lighthouse-core/gather/gatherers/image-elements.js [hunk 2]
```diff
@@ -245,6 +351,9 @@ class ImageElements extends Gatherer {
       const {resourceSize = 0, transferSize = 0} = networkRecord;
       element.resourceSize = Math.min(resourceSize, transferSize);
 
+      if (!element.isInShadowDOM && !element.isCss) {
```
Comment (connorjclark) [inline lighthouse-core/gather/gatherers/image-elements.js:354]: only change

**Comment by connorjclark** [comment]
I'm not sure what further speedups we could manage. If this isn't fast enough, we shouldn't land, because I'm not confident more will be done in the next 3 weeks.

**Comment by patrickhulce** [comment]
> I'm not sure what further speedups we could manage.

I'd vote to cap to the largest N images by display size like we do with optimized images gatherer for natural size or use an explicit time budget (once the gatherer has taken 5s stop resolving the CSS size). The largest images will have the largest CLS impact, so we should still get most of the value here.

**Comment by connorjclark** [comment]
Alright, we could do that in a follow up. How about sort by size + time budget?

**Comment by patrickhulce** [comment]
> How about sort by size + time budget?

SG 👍 and yes definitely followup material :)

**Pull request merged**